src/Features/Core/Portable/Completion/CompletionHelper.cs

```diff
@@ -16,8 +16,8 @@ internal sealed class CompletionHelper
         private static readonly CompletionHelper CaseInsensitiveInstance = new CompletionHelper(isCaseSensitive: false);
 
         private readonly object _gate = new object();
-        private readonly Dictionary<(CultureInfo, string), PatternMatcher> _patternMatcherMap =
-             new Dictionary<(CultureInfo, string), PatternMatcher>();
+        private readonly Dictionary<(CultureInfo, string, bool includeMatchedSpans), PatternMatcher> _patternMatcherMap =
```

Comment:
❕ None of these `PatternMatcher` instances get disposed.

❗️ The `CompletionHelper` itself is a static instance, meaning everything added to this map will hold a strong reference to the `PatternMatcher` for the life of the application.